python/hail/methods/statgen.py

```diff
@@ -2985,13 +2992,15 @@ def ld_prune(ds, r2=0.2, window=1000000, memory_per_core=256):
     :class:`.Table`
         Table of variants after pruning.
     """
-    sites_only_table = _local_ld_prune(ds, r2, window, memory_per_core)
+    check_entry_indexed('ld_prune/call_expr', call_expr)
+    mt = matrix_table_source('ld_prune/call_expr', call_expr).select_entries(__gt=call_expr)
```

Comment:
I think I'd prefer here if you checked to see if it was already a field (you can do this by examining the AST) for now so that we're not doing work when an entry field already exists, but if other methods are just doing this, maybe that's fine here, too.

Reply:
I think other methods all do the check, like in PCA